### PR TITLE
Migrate to hypothesis.strategies.datetimes

### PR DESCRIPTION
**Pull request by himanshukgp**
Regarding #2275
Made required changes in sunpy/net/dataretriever/tests/test_eve.py

### CODE REVIEW

File: sunpy/net/dataretriever/tests/test_eve.py [hunk 1]
```diff
@@ -84,7 +85,11 @@ def test_fido(query):
 
 
 @pytest.mark.remote_data
-@given(time_attr(time=datetimes(timezones=[], max_year=datetime.datetime.utcnow().year, min_year=2010)))
+@given(time_attr(time=datetimes(
+    # timezones=None,
```
Comment (dpshelio) [inline sunpy/net/dataretriever/tests/test_eve.py]: Why is this commented out? is it needed?

Comment (himanshukgp) [reply sunpy/net/dataretriever/tests/test_eve.py]: No it is not needed. I'll remove it in next commit.

**Comment by nabobalis** [comment]
I think the import is used in other files/locations

**Comment by himanshukgp** [comment]
Yes, I am working on it.

**Comment by nabobalis** [comment]
I just fixed the eve test failure that occurred on my machine. Hopefully it will pass!

**Comment by nabobalis** [comment]
@himanshukgp Thanks for this! Sorry for getting a bits handsy on it, I was just thinking of adding this for our next release (0.8.3)!

File: sunpy/net/dataretriever/tests/test_eve.py [hunk 2]
```diff
@@ -25,11 +26,11 @@
      ),
     (TimeRange('2012/5/5', '2012/5/6'),
      'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120505_EVE_L0CS_DIODES_1m.txt',
-     'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120506_EVE_L0CS_DIODES_1m.txt'
+     'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120506_EVE_L0CS_DIODES_1m.txt',
      ),
     (TimeRange('2012/7/7', '2012/7/14'),
      'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120707_EVE_L0CS_DIODES_1m.txt',
-     'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120714_EVE_L0CS_DIODES_1m.txt'
+     'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/L0CS/SpWx/2012/20120714_EVE_L0CS_DIODES_1m.txt',
```
Comment (nabobalis) [inline sunpy/net/dataretriever/tests/test_eve.py]: The extra commas I've added here probably need removing.

Comment (dpshelio) [reply sunpy/net/dataretriever/tests/test_eve.py]: Commas at the end of a container don't add anything. They are though a good practice for version control if you think that list, tuple, ... will grow over time, so there's not 2 lines changed, but only 1 new addition each time. (Probably not the case here in the inner tuple, but would do on the outer list).

**Comment by nabobalis** [comment]
So the test failure happens since the figure envs have a hard pinned hypothesis version that is before they changes the API.

So those versions need updating in the environment files under `sunpy/tests/figure_*`. It should not cause a hashchange.

**Comment by himanshukgp** [comment]
hypothesis=3.42.1=py27_1
hypothesis=3.42.1=py35_1
I have updated hypothesis as above in two respective files in sunpy/tests/figure_*
Is this what you meant ?

**Comment by nabobalis** [comment]
@himanshukgp Yes that would be exactly right.

**Comment by himanshukgp** [comment]
@nabobalis I made the required change but made some mistakes in the process. Please guide me on how should I move forward with this PR. 
Thankyou

**Comment by nabobalis** [comment]
Mistakes? The PR looks to be ready. I'll go over it again once more.

**Comment by himanshukgp** [comment]
Same commits have been pushed twice here.

**Comment by nabobalis** [comment]
We can clean that up when we merge it.

**Comment by nabobalis** [comment]
The CI failed due to doctests.

**Pull request merged**